modules/ai/assets/js/editor/pages/form-layout/index.js

```diff
@@ -141,6 +160,30 @@ const FormLayout = ( { onClose, onInsert, onData, onSelect, onGenerate, DialogHe
 		};
 	};
 
+	/**
+	 * @param {Attachment[]} items
+	 */
+	const onAttach = ( items ) => {
+		items.forEach( ( item ) => {
+			if ( ! attachmentsTypes[ item.type ] ) {
+				throw new Error( `Invalid attachment type: ${ item.type }` );
+			}
+
+			const typeConfig = attachmentsTypes[ item.type ];
+
+			if ( ! item.previewHTML && typeConfig.previewGenerator ) {
+				typeConfig.previewGenerator( item.content ).then( ( html ) => {
+					item.previewHTML = html;
+
+					setAttachments( () => [ item ] );
+				} );
+			}
+		} );
+
+		setAttachments( items );
```

Comment:
something seems weird with those 2 `setAttachments`
I think I'm missing something

Reply:
It's an async process generating the thumbnail